pkg/controller/replication/replication_controller.go

```diff
@@ -195,6 +198,13 @@ func (rm *ReplicationManager) Run(workers int, stopCh <-chan struct{}) {
 // getPodController returns the controller managing the given pod.
 // TODO: Surface that we are ignoring multiple controllers for a single pod.
 func (rm *ReplicationManager) getPodController(pod *api.Pod) *api.ReplicationController {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	controller, cached := rm.lookupCache.GetCachedRC(pod)
+	if cached && rm.isCacheValid(pod, controller) {
```

Comment:
Note - that this is changing the semantics that we had before.
Basically the problem is when we have multiple RCs matching a given pod.

So when we create a new RC that is also matching a given pod:
- in the new approach it will just be ignored (assuming that the previously matching RC didn't changed)
- in the currently existing approach, we may now choose the new RC (depending on the sorting order)

To be clear - I'm not saying the new approach is better or worse - it's just different and we should be explicit if we decide to change it.

So maybe the way to go is to:
- proceed with #21674 for 1.2 (since we want as few changes as possible now)
- proceed with this PR after 1.2

@lavalamp - thoughts?